### PR TITLE
Added metadata to legends.py

### PR DESCRIPTION
**Pull request by chinmaychahar**
Added metadata to legends.py in examples/models/legends.py

- [x] issues: addresses #11765

### CODE REVIEW

**Comment by bryevdv** [comment]
Hi @chinmaychahar FYI it is not necessary to close PRs and open new ones to make changes. You can make and commit new changes in the existing branch, and then `git push` them from the same branch. Then GitHub will update the existing PR. This is preferable, so that that there is a chain of continuity for reviews.

**Comment by chinmaychahar** [comment]
Hey @bryevdv, thanks for mentioning that. I actually had to delete the previous fork because there were some issues with the local setup. That's why deleting and creating the new fork closed the PR and I had to reopen a new one. Hope that's okay

**Comment by bryevdv** [comment]
@chinmaychahar 👍 no worries, FYI @tcmetzger has indicated he should be able to review this PR by the end of the week

File: examples/models/legends.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+'''This example shows different ways to place legends on a graph using LegendLocation. 
+The legend can be placed at defined axes, fixed locations, name locations and side panels. 
+The legend can be defined in terms of title, shape and orientation. This graphs shows sin and cos graph. 
+.. bokeh-example-metadata::
```
Comment (tcmetzger) [inline examples/models/legends.py:4]: Please add a blank line above the `.. bokeh-example-metadata::` section

File: examples/models/legends.py [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+    :apis: bokeh.core.enus.LegendLocation, bokeh.io.show, bokeh.models.Circle, bokeh.models.ColumnDataSource, bokeh.models.DataRange1d, bokeh.models.Plot
```
Comment (tcmetzger) [inline examples/models/legends.py]: How about adding bokeh.models.Legend to this list?

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Thank you! I only have two small requests!

File: examples/models/legends.py [hunk 3]
```diff
@@ -1,3 +1,11 @@
+'''This example shows different ways to place legends on a graph using LegendLocation. 
+The legend can be placed at defined axes, fixed locations, name locations and side panels. 
+The legend can be defined in terms of title, shape and orientation. This graphs shows sin and cos graph.
```
Comment (tcmetzger) [inline examples/models/legends.py]:
```suggestion
The legend can be defined in terms of title, shape, and orientation. This graph shows sin and cos plots. 
```
Please use singular/plural

File: examples/models/legends.py [hunk 4]
```diff
@@ -1,3 +1,11 @@
+'''This example shows different ways to place legends on a graph using LegendLocation.
```
Comment (tcmetzger) [inline examples/models/legends.py]: This example is not using LegendLocation to place legends. Rather, it is using the Legend object to place legends in different locations, all of which are included in the enum LegendLocation.

How about:

```suggestion
'''This example shows different ways to place legends on a graph using the different location values included in the enumeration ``LegendLocation``. 
```

File: examples/models/legends.py [hunk 5]
```diff
@@ -1,3 +1,11 @@
+'''This example shows different ways to place legends on a graph using LegendLocation. 
+The legend can be placed at defined axes, fixed locations, name locations and side panels.
```
Comment (tcmetzger) [inline examples/models/legends.py]: This example only demonstrates placing legends in the named locations included in the enumeration. You can just omit this sentence.

```suggestion

```

File: examples/models/legends.py (hunk 2 above)
Comment (tcmetzger) [inline examples/models/legends.py]:
```suggestion
    :apis: bokeh.core.enums.LegendLocation, bokeh.io.show, bokeh.models.Circle, bokeh.models.ColumnDataSource, bokeh.models.DataRange1d, bokeh.models.Plot
```

File: examples/models/legends.py [hunk 6]
```diff
@@ -1,3 +1,11 @@
+'''This example shows different ways to place legends on a graph using LegendLocation. 
+The legend can be placed at defined axes, fixed locations, name locations and side panels. 
+The legend can be defined in terms of title, shape and orientation. This graphs shows sin and cos graph. 
+.. bokeh-example-metadata::
+    :apis: bokeh.core.enus.LegendLocation, bokeh.io.show, bokeh.models.Circle, bokeh.models.ColumnDataSource, bokeh.models.DataRange1d, bokeh.models.Plot
+    :keywords: sin, cos, pi, linspace, LegendLocation
```
Comment (tcmetzger) [inline examples/models/legends.py]: You could add a reference to the user guide here, like:

```suggestion
    :refs: :ref:`ug_styling_plots_legends`
    :keywords: sin, cos, pi, linspace, LegendLocation
```

**Comment by chinmaychahar** [comment]
I've made the changes accordingly. Please let me know if I missed on something

**Comment by bryevdv** [comment]
HI @chinmaychahar this is almost ready, there are just some linter issues to deal with:

```
E        File contains trailing whitespace: examples/models/legends.py, line 1.
E         File contains trailing whitespace: examples/models/legends.py, line 2.
E         File contains trailing whitespace: examples/plotting/tap.py, line 2
```
That last one is not your fault, and is fixed in another PR, but the simplest thing here is probably just to delete the trailing space in your branch. 

```
E         examples/models/legends.py:5:1: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
E         examples/models/legends.py:5:3: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
E         examples/models/legends.py:5:166: E501 Line too long (175 > 165 characters)
E         examples/models/legends.py:6:1: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
E         examples/models/legends.py:6:3: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
```
it looks like some unicode spaces crept in an should be replaces with a basic space character, also the `apis` line is too long. I would suggest removing one or two of the references. Eventually we will hopefully auto-reference everything and won't need these manual references anyway.

**Comment by chinmaychahar** [comment]
@bryevdv Thanks for the review, I made the changes.

**Comment by bryevdv** [comment]
@chinmaychahar There is still a unicode space issue tripping up the linter:
```
FAILED tests/codebase/test_ruff.py::test_ruff - AssertionError: ruff issues:
  examples/models/legends.py:5:1: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
  examples/models/legends.py:5:3: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
  examples/models/legends.py:6:1: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
  examples/models/legends.py:6:3: RUF002 [*] Docstring contains ambiguous unicode character ` ` (did you mean ` `?)
  Found 4 errors.
  [*] 4 potentially fixable with the --fix option.
```

I think if you run `ruff --fix .` in the top level of the repo, it might fix those issues for you (and then you can commit and push the changes that `ruff` made)

**Comment by chinmaychahar** [comment]
@bryevdv there were some rebase issues but I executed `ruff --fix .` this time, like you suggested and hopefully it should it work.

**Review by bryevdv — APPROVED** [review]

**Review by bryevdv — CHANGES_REQUESTED** [review]
@chinmaychahar I was about to merge but then I noticed the unrelated changed to [examples/styling/mathtext/latex_schrodinger.py](https://github.com/bokeh/bokeh/pull/12920/files#diff-887c7c260e38d9157d80345ed69b1aa90dc3e6212fd8cd088a890034bed59da1) are in this PR for some reason. Can you remove that change? We should keep PRs on-topic, also changing those colors will require generating new thumbnails so that should all happen in a dedicated PR.

**Comment by chinmaychahar** [comment]
@bryevdv yes, you're right. I've removed those changes from this branch

File: examples/styling/mathtext/latex_schrodinger.py [hunk 7]
```diff
@@ -45,8 +45,8 @@
     yupper = np.where(y >= E_v, y, E_v)
     ylower = np.where(y <= E_v, y, E_v)
 
-    p.varea(q, yupper, E_v, fill_color="coral")
-    p.varea(q, ylower, E_v, fill_color="orange")
+    p.varea(q, yupper, E_v, fill_color= "coral")
+    p.varea(q, ylower, E_v, fill_color= "orange")
```
Comment (bryevdv) [inline examples/styling/mathtext/latex_schrodinger.py]:
```suggestion
    p.varea(q, yupper, E_v, fill_color="coral")
    p.varea(q, ylower, E_v, fill_color="orange")
```

**Comment by bryevdv** [comment]
@chinmaychahar almost there, the example file still showing up in the PR due to formatting differences from the original. I've left a suggestion.

**Comment by chinmaychahar** [comment]
Done, apologies for the minor errors every time! @bryevdv

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
No apologies necessary @chinmaychahar OSS is always an iterative process, thanks for the PR!

**Pull request merged**